### PR TITLE
Implement line hit testing and hover inspection along lines

### PR DESCRIPTION
**Pull request by fpliger**
Issues: #1486

### CODE REVIEW

File: bokehjs/src/coffee/tool/inspectors/hover_tool.coffee [hunk 1]
```diff
@@ -87,11 +86,14 @@ define [
 
       xmapper = frame.get('x_mappers')[renderer.mget('x_range_name')]
       ymapper = frame.get('y_mappers')[renderer.mget('y_range_name')]
-      x = xmapper.map_from_target(vx)
-      y = ymapper.map_from_target(vy)
+      # get x, y values from the rendered glyph
+      geom_x = xmapper.map_from_target(vx)
+      geom_y = ymapper.map_from_target(vy)
```
Comment (mattpap) [inline bokehjs/src/coffee/tool/inspectors/hover_tool.coffee]: Where `geom_{x,y}` are used?

Comment (fpliger) [inline bokehjs/src/coffee/tool/inspectors/hover_tool.coffee]: they are not used (yet). No sense to use it if I'm following a wrong path. If, instead, what I've commented makes sense I'd suggest adding `geom_x` and `geom_y` or something similar [here](https://github.com/bokeh/bokeh/blob/feature/hover_along_lines/bokehjs/src/coffee/tool/inspectors/hover_tool.coffee#L134-L141)

**Comment by bryevdv** [comment]
@fpliger after some thought, I think this is the best way to proceed:

* point selections against line(s) need to do a real point/line distance computation (I will dig up some old code)
* for `multi_line` and `patches` the selection should be the standard list of indices
* for `line` and `patch` there will be a special case: if the single line or patch is selected, allot selection to be `True`

**Comment by damianavila** [comment]
This is closed according to [BEP 1](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management#pull-requests). I will not delete the branch so you can iterate on it and reopen the PR when it is ready again for review.

**Pull request closed without merging**